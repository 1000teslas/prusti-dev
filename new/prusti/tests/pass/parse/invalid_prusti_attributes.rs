#![feature(register_tool)]
#![register_tool(prusti)]
#![feature(stmt_expr_attributes)]
#![feature(custom_inner_attributes)]

#[prusti::requires::something(true)]
fn test1() {}

#[prusti::ensures2(true)]
fn test2() {}

fn test3() {
    for _ in 0..2 {
        #![prusti::invariant2(true)]
    }
}

fn test4() {
    for _ in 0..2 {
        #![prusti::invariant::something(true)]
    }
}

#[prusti::requires(true)]
#[prusti::ensures(true)]
fn test5() {
    for _ in 0..2 {
        #![prusti::invariant2(true)]
    }
}

#[prusti::requires(true)]
#[prusti::ensures(true)]
fn test6() {
    for _ in 0..2 {
        #![prusti::invariant::something(true)]
    }
}

fn main() {}
