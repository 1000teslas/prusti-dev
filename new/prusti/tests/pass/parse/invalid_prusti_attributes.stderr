error: "unexpected Prusti attribute; expected `prusti::requires` or `prusti::ensures`"
 --> $DIR/invalid_prusti_attributes.rs:6:1
  |
6 | #[prusti::requires::something(true)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: "unexpected Prusti attribute; expected `prusti::requires` or `prusti::ensures`"
 --> $DIR/invalid_prusti_attributes.rs:9:11
  |
9 | #[prusti::ensures2(true)]
  |           ^^^^^^^^

error: "unexpected Prusti attribute; expected `prusti::invariant`"
  --> $DIR/invalid_prusti_attributes.rs:12:1
   |
12 | / fn test3() {
13 | |     for _ in 0..2 {
14 | |         #![prusti::invariant2(true)]
15 | |     }
16 | | }
   | |_^

error: "unexpected Prusti attribute; expected `prusti::invariant`"
  --> $DIR/invalid_prusti_attributes.rs:18:1
   |
18 | / fn test4() {
19 | |     for _ in 0..2 {
20 | |         #![prusti::invariant::something(true)]
21 | |     }
22 | | }
   | |_^

error: "unexpected Prusti attribute; expected `prusti::invariant`"
  --> $DIR/invalid_prusti_attributes.rs:28:20
   |
28 |         #![prusti::invariant2(true)]
   |                    ^^^^^^^^^^

error: "unexpected Prusti attribute; expected `prusti::invariant`"
  --> $DIR/invalid_prusti_attributes.rs:36:9
   |
36 |         #![prusti::invariant::something(true)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to 6 previous errors

